app/uk/gov/hmrc/apisubscriptionfields/repository/SubscriptionFieldsIdRepository.scala

```diff
@@ -17,40 +17,40 @@
 package uk.gov.hmrc.apisubscriptionfields.repository
 
 import java.util.UUID
+import javax.inject.{Inject, Singleton}
 
+import com.google.inject.ImplementedBy
 import play.api.libs.json._
 import play.modules.reactivemongo.MongoDbConnection
 import reactivemongo.api.DB
 import reactivemongo.api.commands.WriteResult
-import reactivemongo.api.indexes.IndexType.Ascending
 import reactivemongo.api.indexes.Index
+import reactivemongo.api.indexes.IndexType.Ascending
 import reactivemongo.bson.{BSONDocument, BSONObjectID}
-import uk.gov.hmrc.apisubscriptionfields.model.{ApiSubscription, MongoFormatters}
+import uk.gov.hmrc.mongo.ReactiveRepository
 import uk.gov.hmrc.mongo.json.ReactiveMongoFormats
-import uk.gov.hmrc.mongo.{ReactiveRepository, Repository}
 
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
-trait SubscriptionFieldsIdRepository extends Repository[ApiSubscription, BSONObjectID] {
+@ImplementedBy(classOf[SubscriptionFieldsIdMongoRepository])
+trait SubscriptionFieldsIdRepository {
 
-  def save(subscription: ApiSubscription): Future[Unit]
+  def save(subscription: SubscriptionFields): Future[Unit]
 
-  def fetchById(id: String): Future[Option[ApiSubscription]]
-  def fetchByFieldsId(fieldsId: UUID): Future[Option[ApiSubscription]]
+  def fetchById(id: String): Future[Option[SubscriptionFields]]
+  def fetchByFieldsId(fieldsId: UUID): Future[Option[SubscriptionFields]]
 
   def delete(id: String): Future[Unit]
```

Comment:
You could return `Future[Boolean]`:
- `true` if the item was deleted
- `false` if nothing was deleted